addons/point_of_sale/models/pos_session.py

```diff
@@ -256,9 +256,18 @@ def action_pos_session_open(self):
             session.write(values)
         return True
 
+    def _check_if_draft_orders(self):
+        for session in self:
+            for order in session.order_ids:
+                if order.state == "draft":
+                    return True
+            return False
+
     def action_pos_session_closing_control(self):
         self._check_pos_session_balance()
         for session in self:
+            if self._check_if_draft_orders():
```

Comment:
replace this function by "any(order.state == 'draft' for order in session.order_ids)"